### PR TITLE
[apps] Docker only show current containers

### PR DESCRIPTION
**Pull request by Npeca75**
RRDs associated with APPs are created when new APP is discovered, but newer deleted until RRD purge period. This is normally few months/year 
one of problematic APPs is Docker
Problem is that LibreNMS will stack these inactive RRDs on graph, so after few "lets try some containers" it is getting messy. Dozen of inactive RRDs are shown in cumulative graph

This PR add filter ability to filter out unused RRDs matching against active APPs in ApplicationMetrics table

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
(https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Npeca75** [comment]
lnms1
lnms2

Before

**Comment by Npeca75** [comment]
lnms3
lnms4

... after

File: LibreNMS/Data/Store/Rrd.php [hunk 1]
```diff
@@ -511,7 +515,20 @@ public function getRrdApplicationArrays($device, $app_id, $app_name, $category =
             }
         }
 
-        return $entries;
+        //apply filter
+        if ($filter == 'metrics' && $entries) {
+           foreach ($entries as $rrdName) {
+                $cnt = ApplicationMetrics::where('app_id', $app_id)
+                    ->where('metric', 'like', $rrdName . '%')
+                    ->count();
```
Comment (Jellyfrog) [inline LibreNMS/Data/Store/Rrd.php:523]:
```suggestion
                    ->exists();
```

File: phpstan-baseline.neon [hunk 2]
```diff
@@ -8290,6 +8290,11 @@ parameters:
 			count: 1
 			path: app/Models/Application.php
 
+		-
+			message: "#^PHPDoc type array of property App\\\\Models\\\\ApplicationMetrics\\:\\:\\$primaryKey is not covariant with PHPDoc type string of overridden property Illuminate\\\\Database\\\\Eloquent\\\\Model\\:\\:\\$primaryKey\\.$#"
```
Comment (Jellyfrog) [inline phpstan-baseline.neon:8294]: Not allowed to add stuff to this file

File: LibreNMS/Data/Store/Rrd.php (hunk 1 above)
Comment (Npeca75) [reply LibreNMS/Data/Store/Rrd.php]: done, changed as suggested

File: phpstan-baseline.neon (hunk 2 above)
Comment (Npeca75) [reply phpstan-baseline.neon:8294]: hi @Jellyfrog 
if you look at table "application_metrics" you will see that there is composite primary key
`UNIQUE INDEX `application_metrics_app_id_metric_unique` (`app_id`, `metric`) USING BTREE`

only existing  App/Models with two primaryKeys is UserPref.php
and it is listed in "phpstan-baseline.neon" exactly from this reason
`    protected $primaryKey = ['user_id', 'pref'];`

without this, error is thrown on check
so i make same as UserPref.php

if you have any other suggestion ? how to solve this other way ?

Comment (Npeca75) [reply phpstan-baseline.neon:8294]: error is about $primaryKey is array in my case

`PHPDoc type array of property App\Models\ApplicationMetrics::$primaryKey is not covariant with PHPDoc type string of overridden property Illuminate\Database\Eloquent\Model::$primaryKey. `

**Review by murrant — CHANGES_REQUESTED** [review]
NACK from me.

There is a new data field on the applications table.  Just store the list of current application items there and use that to generate the graphs.

**Comment by Npeca75** [comment]
hi @murrant 
> There is a new data field on the applications table. 

ok, first time dealing with this field
done as in ZFS
hope it is OK now

**Review by murrant — APPROVED** [review]
Much better! Thanks @Npeca75

**Review by murrant — APPROVED** [review]

**Review by murrant — APPROVED** [review]

**Pull request merged**